### PR TITLE
Fix consumer connection issue

### PR DESCRIPTION
**Pull request by arszen123**
In the #1382 PR, I made a mistake that has performance implications.

When a `KafkaJSConnectionError` (ex. Connection timeout) is thrown during the fetching + processing, the consumer restarts. I think it is not supposed to restart but retry the fetch.
Based on the tests it's only supposed to restart when there are no available brokers (ex. the cluster is disconnected) https://github.com/tulios/kafkajs/blob/master/src/consumer/__tests__/connection.spec.js#L62-L103

Based on the above-mentioned I made a modification so the consumer will retry on the `KafkaJSConnectionError` error, and restarts when there are no available brokers.

### CODE REVIEW

File: src/consumer/fetchManager.spec.js [hunk 1]
```diff
@@ -94,4 +94,10 @@ describe('FetchManager', () => {
       await expect(fetchManagerPromise).rejects.toThrow('Node not found')
     })
   })
+
+  it('should throw an error when there are no avaiblae brokers', async () => {
```
Comment (michie1) [inline src/consumer/fetchManager.spec.js:98]: `avaiblae` -> `available`?

**Comment by ekanth** [comment]
@arszen123 - Thanks for this fix. And I think this fix is also needed to handle the scenario where `KafkaJSConnectionClosedError` causes an infinite loop when no brokers are reachable (reproducible easily with a single broker docker setup). Same as this [issue](https://github.com/tulios/kafkajs/issues/1384#issuecomment-1160598427) but caused by `KafkaJSConnectionClosedError`. @Nevon @tulios - appreciate if you could review/merge this fix when you get a chance.

File: src/consumer/fetchManager.spec.js (hunk 1 above)
Comment (Saeger) [inline src/consumer/fetchManager.spec.js:98]:
```suggestion

  it('should throw an error when there are no brokers available', async () => {
```

**Comment by kosarajupavankumar** [comment]
ERROR [Connection] Connection error: getaddrinfo EAI_AGAIN edEVENTHUBS01.servicebus.windows.net {"timestamp":"2023-08-14T16:31:54.290Z","logger":"kafkajs","broker":"EVENTHUBS01.servicebus.windows.net:9093","clientId":"microservices/loan/consumer","stack":"Error: getaddrinfo EAI_AGAIN edEVENTHUBS01.servicebus.windows.net\n at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:107:26)\n at GetAddrInfoReqWrap.callbackTrampoline (node:internal/async_hooks:130:17)"}

ERROR [Producer] Failed to send messages: Connection error: Client network socket disconnected before secure TLS connection was established {"timestamp":"2023-08-14T16:31:10.114Z","logger":"kafkajs","retryCount":0,"retryTime":249}

@Nevon and @arszen123 after using the latest version of the 2.2.4 still we are getting the connection errors from the kafka side

**Pull request merged**